File: customer_app/lib/models/address.dart
class Address {
  final int id;
  final String name;
  final String mobile;
  final String villageTown;
  final String? house;
  final String? landmark;
  final String pincode;
  final bool isDefault;

  Address({
    required this.id,
    required this.name,
    required this.mobile,
    required this.villageTown,
    this.house,
    this.landmark,
    required this.pincode,
    this.isDefault = false,
  });

  factory Address.fromJson(Map<String, dynamic> json) {
    return Address(
      id: json['id'],
      name: json['name']?.toString() ?? '',
      mobile: json['mobile']?.toString() ?? '',
      villageTown: json['village_town']?.toString() ?? '',
      house: json['house']?.toString(),
      landmark: json['landmark']?.toString(),
      pincode: json['pincode']?.toString() ?? '',
      isDefault: json['is_default'] ?? false,
    );
  }

  Map<String, dynamic> toJson() => {
        'name': name,
        'mobile': mobile,
        'village_town': villageTown,
        if (house != null && house!.isNotEmpty) 'house': house,
        if (landmark != null && landmark!.isNotEmpty) 'landmark': landmark,
        'pincode': pincode,
        'is_default': isDefault,
      };

  String get oneLine => [house, villageTown, landmark, pincode]
      .where((e) => e != null && e.isNotEmpty)
      .join(', ');
}

File: backend/app/models/address.py
from sqlalchemy import Column, Integer, String, Float, Boolean, ForeignKey
from sqlalchemy.orm import relationship

from app.core.database import Base


class Address(Base):
    __tablename__ = "addresses"

    id = Column(Integer, primary_key=True, index=True)
    customer_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    name = Column(String(100), nullable=False)
    mobile = Column(String(15), nullable=False)
    village_town = Column(String(150), nullable=False)
    house = Column(String(150), nullable=True)
    landmark = Column(String(150), nullable=True)
    pincode = Column(String(10), nullable=False, index=True)
    lat = Column(Float, nullable=True)
    lng = Column(Float, nullable=True)
    is_default = Column(Boolean, default=False)

    customer = relationship("User", back_populates="addresses")


class DeliveryZone(Base):
    """Local delivery-area whitelist (requirement #26) — admin managed."""
    __tablename__ = "delivery_zones"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(150), nullable=False)
    pincode = Column(String(10), nullable=False, index=True)
    is_active = Column(Boolean, default=True)
